R/module_teal_data.R

```diff
@@ -74,19 +91,58 @@ srv_teal_data <- function(id,
       validate_shiny_silent_error = validate_shiny_silent_error
     )
 
-    .fallback_on_failure(
-      this = data_validated,
-      that = data,
-      label = sprintf("Data element '%s' for module '%s'", id, modules$label)
-    )
+    out <- reactiveVal(NULL)
+    observeEvent(data_validated(), {
+      if (inherits(data_validated(), "teal_data")) {
+        is_transformer_failed[[id]] <- FALSE
+        if (!identical(data_validated(), out())) {
+          out(data_validated())
+        }
+      } else {
+        is_transformer_failed[[id]] <- TRUE
+      }
```

Comment:
Maybe let's keep logging information the same way we had it before in `.failure_on_fallback`? I think this was useful

```suggestion
      if (inherits(data_validated(), "teal_data")) {
        is_transformer_failed[[id]] <- FALSE
        if (!identical(data_validated(), out())) {
          out(data_validated())
          logger::log_debug(sprintf("Data element '%s' for module '%s' evaluated successfully.", id, modules$label))
        }
      } else {
        is_transformer_failed[[id]] <- TRUE
        logger::log_debug(sprintf("Data element '%s' for module '%s' failed. Falling back to previous data.", id, modules$label))
      }
```

Reply:
👍 edited the suggestion